pkgs/development/ocaml-modules/opam-publish/default.nix

```
@@ -0,0 +1,34 @@
+{ lib, buildDunePackage, fetchFromGitHub
+, cmdliner, lwt_ssl, opam-core, opam-format, opam-state, github, github-unix
+}:
+
+buildDunePackage rec {
+  pname = "opam-publish";
+  version = "2.2.0";
+
+  src = fetchFromGitHub {
+    owner = "ocaml-opam";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-FNAWok5tjTOwwpNZ0Xcu9E/R8iXStZqCk/Vqdf9l+zs=";
+  };
+
+  duneVersion = "3";
+
+  propagatedBuildInputs = [
```

Are you sure this must be propagated?

---

So you'd suggest this, then?
```suggestion
  buildInputs = [
```
It looks like things build just fine with that instead of `propagatedBuildInputs`. To be honest, I don't really know why, I just based my packages on other OCaml packages and it looked like a lot of them used `propagatedBuildInputs` so I did that too. You'd say to go for `buildInputs` by default and `propagatedBuildInputs` only if necessary?

---

Yes, `buildInputs` is a good default.

---

Done in a2e2e80c3d3 (forced-pushed onto my branch).